src/java/org/apache/cassandra/transport/CBUtil.java

```diff
@@ -478,7 +487,37 @@ public static void writeValue(ByteBuffer bytes, ByteBuf cb)
         cb.writeInt(remaining);
 
         if (remaining > 0)
-            cb.writeBytes(bytes.duplicate());
+        {
+            cb.writeBytes(MemoryUtil.duplicateHeapByteBuffer(bytes, getLocalBuffer()));
+        }
+    }
+
+    public static void writeValue(ByteBuffer src, ByteBuf dest)
+    {
+        if (src == null)
+        {
+            dest.writeInt(-1);
+            return;
+        }
+
+        int length = src.remaining();
+        dest.writeInt(length);
+
+        if (src.hasArray())
+        {
+            byte[] array = src.array();
```

Comment:
can you add a comment explaining why we do this when direct duplicates the buffer?  You found a JDK bug, we should expose that to anyone reading so they know that there may be dragons to replicate

Reply:
Is this comment descriptive enough?

Reply:
can you flesh this out more?  Scott showed you the bug.

it is also not an "improper" way, it was a bug in CMS